### ISSUE TITLE
passing string for double-valued option doesn't error

### ISSUE THREAD

**Issue by lkapelevich** [ERGO-Code/HiGHS#1250]
With 
```
HighsStatus stat = highs.setOptionValue("time_limit", "hi");
```
I get that stat is `kOk`.

**Comment by jajhall** [comment]
We allow users to pass strings (such as "3.6e+3" for the time limit) in any call to `Highs::setOptionValue`, and assumed that they would pass sensible values. However, there's the danger of the user causing unintended behaviour, as "hi" is converted to a `double` value of 0.

Once [fix-1250](https://github.com/ERGO-Code/HiGHS/tree/fix-1250) is pushed to latest, 
- string values for non-string options have leading/trailing blanks trimmed
- string values for `double` options are illegal if they contain characters other than "+-.0123456789eE"

**Comment by jajhall** [comment]
Fixed no #1252 is merged into latest
